### PR TITLE
Loosen return type guidelines for local functions

### PR DESCRIPTION
**Pull request by natebosch**
Local named functions do infer a return type in current Dart versions.

In our recommendation to annotate the return type of functions change
the set of functions the guideline applies to from "named" functions to
"non-local" functions, anonymous function expressions are also local to
a method body.

### CODE REVIEW

**Review by munificent — APPROVED** [review]
Good catch!

File: src/content/effective-dart/design.md
Comment (lrhn) [inline src/content/effective-dart/design.md:1204]: Maybe "top-level functions, static functions and instance methods".

I don't usually call static functions "methods", although I guess "static method" is commonly used for them. (I think it's a Java way of phrasing it, which began because they don't/didn't have plain functions at all.)

**Review by lrhn — APPROVED** [review]

**Review by MaryaBelanger — APPROVED** [review]
Lgtm, feel free to wait for Parker's review or merge now that you have approval. Thanks!

**Comment by natebosch** [comment]
@MaryaBelanger - I don't have authorization to merge in this repo. Can you merge for me?

**Comment by lrhn** [comment]
Me neither. @munificent?

**Review by parlough — APPROVED** [review]
Thanks for this update!

**Pull request merged**
